### ISSUE TITLE
Preflight reports failed scorecard-sdk tests as "errors" and not as "failed" if there is a non-zero error field

### ISSUE THREAD

**Issue by tkrishtop** [redhat-openshift-ecosystem/openshift-preflight#326]
## Bug Description

I'm testing operator testpmd:
bundle image: `quay.io/rh-nfv-int/testpmd-operator-bundle:v0.2.9`
index image: `quay.io/tkrishtop/index-testpmd-operator-bundle:v0.2.9`

[The command `operator-sdk scorecard --selector=suite=olm`](https://www.distributed-ci.io/files/405aff27-085a-4274-8e09-25ef1261f9c9) reports several failed tests: olm-crds-have-validation-test, olm-status-descriptors-test, olm-bundle-validation-test, olm-crds-have-resources-test, and olm-spec-descriptors-test.

[The command `preflight operator check`](https://www.distributed-ci.io/files/bb21da72-e57d-4596-a2bf-d4e493273011) reports 2 passed tests: ScorecardBasicSpecCheck and DeployableByOLM, and 2 errors: ScorecardOlmSuiteCheck and ValidateOperatorBundle. 

I.e. some _failed_ tests are reported as _error_ by preflight. This happens each time when operator-sdk produces a non-zero error field for the failed test. 

It seems like this behaviour is a [part of the old workaround on the preflight side](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/internal/engine/operatorsdk.go#L65-L72). 
The problem with that behaviour is that it's inconsistent - some failed operator-sdk tests go into "failed" (if they have nothing in the "error" field), and some go into "errors" (if they have a nonzero "error" field).

## Version and Command Invocation
v1.0.3

## Steps to Reproduce:
1) Operator to test:
bundle image: `quay.io/rh-nfv-int/testpmd-operator-bundle:v0.2.9`
index image: `quay.io/tkrishtop/index-testpmd-operator-bundle:v0.2.9`

2) Operator-sdk command:
```
{{ operator_sdk_tool_path }} scorecard \
--output json \
--selector=suite=olm \
--kubeconfig "{{ kubeconfig_path }}" \
--namespace scorecard-testing \
--service-account default \
--config {{ scorecard_config_path }} \
--verbose \
--wait-time 300s \
"{{ scorecard_operator_dir }}"
```

olm-crds-have-validation-test, olm-status-descriptors-test, olm-bundle-validation-test, olm-crds-have-resources-test, and olm-spec-descriptors-test have "fail" status.

3) Preflight command:
```
podman run \
-it \
--rm \
--pull=always \
--name preflight \
--privileged \
-e PFLT_JUNIT=true \
-e PFLT_ARTIFACTS=/artifacts \
-e PFLT_LOGFILE=/artifacts/preflight.log \
-e PFLT_LOGLEVEL=trace \
-e KUBECONFIG=/kubeconfig \
-e PFLT_INDEXIMAGE={{ OO_INDEX }} \
-e PFLT_NAMESPACE=preflight-testing \
-e PFLT_SERVICEACCOUNT=default \
-v {{ preflight_kubeconfig }}/kubeconfig:/kubeconfig \
-v {{ preflight_artifacts }}:/artifacts \
{{ preflight_version }} \
check operator \
{{ operator.image }}
```

ScorecardOlmSuiteCheck and ValidateOperatorBundle tests are in "errors", not in "failed".

4) Here are logs: 
[all log files](https://www.distributed-ci.io/jobs/6e57d051-636a-40f0-8846-b7ba277db5a6/files), 
[results.json](https://www.distributed-ci.io/files/bb21da72-e57d-4596-a2bf-d4e493273011)
[scorecard logs](https://www.distributed-ci.io/files/405aff27-085a-4274-8e09-25ef1261f9c9)

## Expected Result
If the test has a "fail" status in operator-sdk, it should be in "failed" for preflight as well.

## Actual Result
Some failed operator-sdk tests go into "failed" (if they have nothing in the "error" field), and some go into "errors" (if they have a nonzero "error" field).

olm-crds-have-validation-test, olm-status-descriptors-test, olm-bundle-validation-test, olm-crds-have-resources-test, and olm-spec-descriptors-test have "fail" status.

ScorecardOlmSuiteCheck and ValidateOperatorBundle tests are in "errors", not in "failed".

**Comment by tkrishtop** [comment]
cc @samira-barouti

**Comment by bcrochet** [comment]
This will likely be fixed by #334

**Comment by bcrochet** [comment]
Marking this closed since #334 landed. Please reopen if not solved by it.

**Comment by tkrishtop** [comment]
Confirm that the issue is fixed in preflight 1.0.5: https://www.distributed-ci.io/jobs/cbfc5b94-654e-4a27-9494-a78051fa891b/files 
Thank you! :)
